tests/test_utils.go

```diff
@@ -0,0 +1,159 @@
+package tests
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+	"github.com/tablelandnetwork/basin-storage/pkg/ethereum"
+	"github.com/textileio/go-tableland/pkg/wallet"
+	"golang.org/x/crypto/sha3"
+)
+
+const functionsPort = "8293"
+
+func uploadRandomBytesToGCS(t *testing.T, data []byte, bucketName, objectName string) {
+	ctx := context.Background()
+
+	// Create a client
+	client, err := storage.NewClient(ctx)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, client.Close())
+	}()
+
+	// Get a handle to the bucket and the object
+	bucket := client.Bucket(bucketName)
+	object := bucket.Object(objectName)
+
+	// Create a writer and write the random bytes
+	wc := object.NewWriter(ctx)
+	_, err = wc.Write(data)
+	require.NoError(t, err)
+	require.NoError(t, wc.Close())
+}
+
+func deleteObjectFromGCS(t *testing.T, bucketName, objectName string) {
+	ctx := context.Background()
+
+	// Create a client
+	client, err := storage.NewClient(ctx)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, client.Close())
+	}()
+
+	// Get a handle to the bucket and the object
+	bucket := client.Bucket(bucketName)
+	object := bucket.Object(objectName)
+
+	// Delete the object
+	err = object.Delete(ctx)
+	require.NoError(t, err)
+}
+
+func setupDB(t *testing.T, db *sql.DB) {
+	_, err := db.Exec("CREATE DATABASE IF NOT EXISTS basin_test")
+	require.NoError(t, err)
+
+	_, err = db.Exec(
+		`CREATE TABLE IF NOT EXISTS namespaces
+	(
+		id     BIGSERIAL   PRIMARY KEY,
+		name   VARCHAR(32) UNIQUE NOT NULL,
+		owner  BYTEA NOT NULL,
+		last_export TIMESTAMPTZ
+	)`)
+	require.NoError(t, err)
+
+	_, err = db.Exec("INSERT INTO namespaces (name, owner) VALUES ('esfbmltndstj', 'test_owner')")
+	require.NoError(t, err)
+
+	_, err = db.Exec(
+		`CREATE TABLE IF NOT EXISTS jobs
+		(
+			id BIGSERIAL PRIMARY KEY,
+			ns_id BIGINT,
+			cid BYTEA NOT NULL,
+			relation TEXT NOT NULL,
+			activated TIMESTAMP,
+			CONSTRAINT fk_namespace
+			FOREIGN KEY(ns_id)
+			REFERENCES namespaces(id)
+		)`)
+	require.NoError(t, err)
+}
+
+func insertProcessedJob(t *testing.T, db *sql.DB) cid.Cid {
+	// already processed job with 2 deals
+	cidStr := "bafybeie62jygi4bi5m3yd43ttppkn5velvck5yutckggp6iwdmkkowyhxy"
```

Comment:
so this is already in web3 storage?

Reply:
yes, it is already stored. we need a CID that is already finalized because it takes ~48h to make deals.